app/models/lookup_keys/lookup_key.rb

```diff
@@ -172,6 +142,10 @@ def overridden?(host)
     lookup_values.find_by_match(host.send(:lookup_value_match)).present?
   end
 
+  def puppet?
```

Comment:
can we change this to something like features? this will not require the parent class to know anything about the child class and you could ask each class for its "features or capabilities"